boards/khadas/edge2/board.yml

```diff
@@ -0,0 +1,5 @@
+board:
+  name: khadas_edge2
```

Comment:
I think we typically wouldn't want the vendor name to prefix the board name, as it's redundant, is it not @nordicjm? 
There are some boards that do it, but it's a minority AFAICS. That being said, "edge2" definitely is a very generic name so it doesn't sound like a bad idea to anticipate and namespace it, it's just that I am not sure what the rule/guideline is here.